code/_helpers/animations.dm

```diff
@@ -98,4 +98,23 @@
 	for(var/i in 2 to segments) //2 because 1 is covered above
 		animate(transform = matrices[i], time = speed)
 		//doesn't have an object argument because this is "Stacking" with the animate call above
-		//3 billion% intentional
\ No newline at end of file
+		//3 billion% intentional
+
+// This proc is used to move an atom to a target loc and then interpolite to give the illusion of sliding from start to end.
+/proc/do_visual_slide(var/atom/movable/sliding, var/turf/from, var/from_pixel_x, var/from_pixel_y, var/turf/target, var/target_pixel_x, var/target_pixel_y, var/center_of_mass)
+	set waitfor = FALSE
+	var/start_pixel_x = sliding.pixel_x - ((target.x-from.x) * world.icon_size)
+	var/start_pixel_y = sliding.pixel_y - ((target.y-from.y) * world.icon_size)
+	// Clear our glide so we don't do an animation when dropped into the target turf.
+	var/old_animate_movement = sliding.animate_movement
+	sliding.animate_movement = NO_STEPS
+	sleep(2) // Due to BYOND being byond, animate_movement has to be set for at least 2 ticks before gliding will be disabled.
```

Comment:
```suggestion
	sleep(2 * world.tick_lag) // Due to BYOND being byond, animate_movement has to be set for at least 2 ticks before gliding will be disabled.
```
I think Kapu said something about how this is more accurate? Since you're pausing for 2 deciseconds, rather than 2 ticks, which is more than is necessary.